spec/rubocop/cop/rspec/sort_metadata_spec.rb

```diff
@@ -23,6 +23,54 @@
     RUBY
   end
 
+  it 'does not register an offense for a symbol metadata before a non-symbol ' \
+     'argument' do
+    expect_no_offenses(<<~RUBY)
+      RSpec.describe 'Something', :z, :a, 'string' do
+      end
+
+      RSpec.describe 'Something', :z, :a, variable, foo: :bar do
+      end
+    RUBY
+  end
+
+  it 'registers an offense only for trailing symbol metadata not in ' \
+     'alphabetical order' do
+    expect_offense(<<~RUBY)
+      RSpec.describe 'Something', :z, :a, 'string', :c, :b do
+                                                    ^^^^^^ Sort metadata alphabetically.
+      end
+    RUBY
+
+    expect_correction(<<~RUBY)
+      RSpec.describe 'Something', :z, :a, 'string', :b, :c do
+      end
+    RUBY
+  end
+
+  it 'registers an offense when a symbol metadata not in alphabetical order ' \
+     'is before a variable argument being the last argument ' \
+     'as it could be a hash' do
+    expect_offense(<<~RUBY)
+      RSpec.describe 'Something', :z, :a, some_hash do
```

Comment:
Now when I look at `some_hash`, did we mean a literal hash? How do we know `some_hash` is hash, and not a symbol?

I have no objections to such "chunky" reordering is in place.

Let's split this into two:

1. Example with a literal hash
```ruby
describe 'Something', :z, :a, { foo: 'bar' } do
```

2. Example with many chunks
```ruby
describe 'Chunks', :z, :a, var, :y, :b, another_var, :x, :c, last_var
```
The current implementation fixed this to:
```ruby
describe 'Chunks', :a, another_var, :b, :c, var, :x, :y, :z, last_var do
```
Which I don't entirely agree with. Variables should retain their positions. Only literal symbols should be rearranged.

Variables not remaining on their places result is a potentially breaking behaviour:
```ruby
RSpec.describe 'Chunks', second_docstring, :z, :a, var do
```
```ruby
RSpec.describe 'Chunks', :a, second_docstring, :z, var do 
```
:boom: "wrong number of arguments (given 3, expected 0..2)".

Reply:
> Now when I look at some_hash, did we mean a literal hash? How do we know some_hash is hash, and not a symbol?

We don't know.

> 2. Example with many chunks
> 
> ```ruby
> describe 'Chunks', :z, :a, var, :y, :b, another_var, :x, :c, last_var
> ```
> 
> The current implementation fixed this to:
> 
> ```ruby
> describe 'Chunks', :a, another_var, :b, :c, var, :x, :y, :z, last_var do
> ```
> 
> Which I don't entirely agree with. Variables should retain their positions. Only literal symbols should be rearranged.
> 
> Variables not remaining on their places result is a potentially breaking behaviour:

Yep agreed.

With the implementation proposed in this pull request, it would fix this to:

```ruby
describe 'Chunks', :z, :a, var, :y, :b, another_var, :c, :x, last_var
```

only the last symbols part is reordered.

> Variables not remaining on their places result is a potentially breaking behaviour:
> 
> ```ruby
> RSpec.describe 'Chunks', second_docstring, :z, :a, var do
> ```
> 
> ```ruby
> RSpec.describe 'Chunks', :a, second_docstring, :z, var do 
> ```
> 
> 💥 "wrong number of arguments (given 3, expected 0..2)".

With the implementation proposed in this pull request, it would fix this to:

```ruby
RSpec.describe 'Chunks', second_docstring, :a, :z, var do 
```

That's what triggered me to open this PR.

Reply:
> it would fix this to:

In a quick test (I've created a spec/a_spec.rb and ran `rubocop -a`), it fixed differently:
```diff
- describe 'Chunks', :z, :a, var, :y, :b, another_var, :c, :x, last_var do
+ describe 'Chunks', :a, another_var, :b, :c, var, :x, :y, :z, last_var do
```

Can you please check?

The rest seems to be 👍

Reply:
I'm sorry I did not get the same results as you. It left the file unchanged.

Here is how I tested it so you can test it on your end:

From my `code/rubocop-rspec` directory:

```shell
git switch sort_metadata_symbols_at_the_end_of_args_list
mkdir ../checking_cop
cd ../checking_cop
bundle init
bundle add rspec
bundle exec rspec --init

bundle add rubocop
echo 'gem "rubocop-rspec", path: "../rubocop-rspec"' >> Gemfile
bundle install
cat <<RUBOCOP_CONF > .rubocop.yml
require:
  - rubocop-rspec
AllCops:
  NewCops: enable
RUBOCOP_CONF

cat <<SPEC > spec/checking_cop_spec.rb
# frozen_string_literal: true

var = :var
another_var = :another_var
last_var = :last_var

describe 'Chunks', :z, :a, var, :y, :b, another_var, :c, :x, last_var do
  it 'is not an empty example group' do
    expect(true).to be_truthy
  end
end
SPEC

bundle exec rubocop --only RSpec/SortMetadata spec/checking_cop_spec.rb
```

And I get the output

```
Inspecting 1 file
.

1 file inspected, no offenses detected
```

I tested swapping the final parameters and running the cop again:

```shell
cat <<SPEC >> spec/checking_cop_spec.rb
describe 'Chunks 2', :z, :a, var, :y, :b, another_var, :x, :c, last_var do
  it 'is not an empty example group' do
    expect(true).to be_truthy
  end
end
SPEC

bundle exec rubocop --only RSpec/SortMetadata spec/checking_cop_spec.rb
```

got output:

```
Inspecting 1 file
C

Offenses:

spec/checking_cop_spec.rb:12:56: C: [Correctable] RSpec/SortMetadata: Sort metadata alphabetically.
describe 'Chunks 2', :z, :a, var, :y, :b, another_var, :x, :c, last_var do
                                                       ^^^^^^

1 file inspected, 1 offense detected, 1 offense autocorrectable
```

Can you please check?